### PR TITLE
Fix #2766: Update contribution.rst to use black

### LINKED ISSUE #2766: Update contributing guidelines to recommend using Black instead of autopep8
The docs, i.e. https://evalai.readthedocs.io/en/latest/contribution.html recommend using autopep8 to format the code while the README has a [`Black`](https://black.readthedocs.io/en/stable/) tag. 
Appropriately replace the autopep8 part with instructions for Black.

### PR DESCRIPTION
**Pull request by nikochiko**
Changes:
* Replace `autopep8` instructions with `black` instructions
* Separate the `git` code block and `black` code block to avoid confusion
* Add instructions for using pre-commit hook.

### CODE REVIEW

**Comment by nikochiko** [comment]
**Preview:**
image

File: docs/source/contribution.rst [hunk 1]
```diff
@@ -37,17 +37,20 @@ branches based off of development and merged into it once stable and
...
        git fetch upstream
        git checkout master
        git merge upstream/master
        git checkout -b your-branch-name
 
+   We highly encourage using `black <http://www.github.com/psf/black>`_
+   to format your code. It sticks to PEP8 and is in line with the rest
+   of the repo. To use black, install it via `pip` and run the following
+   command:
+   ::
```
Comment (Ayukha) [inline docs/source/contribution.rst:51]: @nikochiko  Can you please add the usage of precommits as mentioned by @RishabhJain2018 https://github.com/Cloud-CV/EvalAI/pull/2762#issuecomment-620754957

Comment (nikochiko) [reply docs/source/contribution.rst:51]: But that's strange. The hooks are already set up via `pre-commit` but I didn't get results of flake8 and black didn't format my very inconsistent test file while testing.

Comment (nikochiko) [reply docs/source/contribution.rst:53]: Added! :+1: 
@Ayukha thanks for the pointer.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 repos:
--   repo: https://github.com/ambv/black
+-   repo: https://github.com/psf/black
```
Comment (nikochiko) [inline .pre-commit-config.yaml:2]: I included a related minor change: 
psf/black is the new location of black and now https://github.com/ambv/black redirects there.

**Pull request merged**